src/api/java/ic2/api/energy/prefab/BasicEnergyTile.java

```diff
@@ -220,7 +220,7 @@ public boolean useEnergy(double amount) {
 	 * @return true if energy was transferred
 	 */
 	public boolean charge(ItemStack stack) {
-		if (stack == null || !Info.isIc2Available() || getWorldObj().isRemote) return false;
+		if (stack == null || stack.isEmpty() || !Info.isIc2Available() || getWorldObj().isRemote) return false;
```

Comment:
This isn't the only occurence, I only commented on the first one ;)
Changing the API isn't needed as you said, it will be overwritten by the next update of the API of IC2